lighthouse-core/lib/i18n/i18n.js

```diff
@@ -151,6 +151,24 @@ const formats = {
   },
 };
 
+/**
+ * Function to augment a given locale with additional messages.
+ * @param {LH.Locale} locale
+ * @param {LH.LocaleMessages} messages
+ * @param {string=} dir Optional absolute path to be appended to message keys.
+ */
+function augmentLocale(locale, messages, dir) {
+  if (dir) {
+    for (const [key, message] of Object.entries(messages)) {
+      delete messages[key];
+      const [filename, varName] = key.split(' | ');
+      const relativeDir = path.relative(LH_ROOT, dir).replace(/\\/g, '/');
+      messages[`${path.join(relativeDir, filename)} | ${varName}`] = message;
```

Comment:
is this a lookup we could support instead of altering the key so the existing lookup still works? e.g. do we just need to drop a `node_modules/` from the key when looking up the message?

Reply:
I don't think we can always assume that this will be in `node_modules`, when running the plugin globally for example. Currently our collect strings keys messages relative to the plugin root: `audits/...` for example. With the refactor, I'm planning to append plugin path to the returned locales from `config-plugin.js`, thoughts?

Reply:
> I don't think we can always assume that this will be in `node_modules`, when running the plugin globally for example. Currently our collect strings keys messages relative to the plugin root: `audits/...` for example. With the refactor, I'm planning to append plugin path to the returned locales from `config-plugin.js`, thoughts?

Good point, I forgot about the various ways of using them we have to support. Local plugin development would also be one we wouldn't want to make any harder than it already is :)

Plugin root makes the most sense, then, and that plan sounds great.